docs/src/introduction_to_catalyst/catalyst_for_new_julia_users.md

```diff
@@ -1,181 +1,229 @@
+1. If Catalyst is not already installed on your computer, install it.
+2. Add Catalyst as an available package to your current environment.
+
+Here, while Catalyst has previously been installed on your computer, it has not been added to the new environment you created. To do so, simply run 
+```julia
+using Pkg
+Pkg.add("Catalyst")
+```
+after which Catalyst can be imported through `using Catalyst`. You can get a list of all packages available in your current environment using:
+```julia
+Pkg.status()
+```
+
+So, why is this required, and why cannot we simply import any package installed on our computer? The reason is that most packages depend on other packages, and these dependencies may be restricted to only specific versions of these packages. This creates complicated dependency graphs that restrict what versions of what packages are compatible with each other. When you use `Pkg.add("PackageName")`, only a specific version of that package is actually added (the latest possible version as permitted by the dependency graph). Here, Julia environments both define what packages are available *and* their respective versions (these versions are also displayed by the `Pkg.status()` command). By doing this, Julia can guarantee that the packages (and their versions) specified in an environment are compatible with each other.
+
+The reason why all this is important is that it is *highly recommended* to, for each project, define a separate environment. To these, only add the required packages. General-purpose environments with a large number of packages often produce package-incompatibility issues. While these might not prevent you from installing all desired package, they often mean that you are unable to use the latest version of some packages.
+
+!!! note
+    A not-infrequent cause for reported errors with Catalyst (typically the inability to replicate code in tutorials) is package incompatibilities in large environments preventing the latest version of Catalyst from being installed. Hence, whenever an issue is encountered, it is useful to run `Pkg.status()` to check whenever the latest version of Catalyst is being used.
+
+Some additional useful Pkg commands are:
+- `Pk.rm("PackageName")` removes a package from the current environment.
+- `Pkg.update(PackageName")`: updates the designated package.
 
+!!! note
+    A useful feature of Julia's environment system is that enables the exact definition of what packages and versions were used to execute a script. This supports e.g. reproducibility in academic research. Here, by providing the corresponding Project.toml and Manifest.toml files, you can enable someone to reproduce the exact program just to perform some set of analyses.
+    
```

Comment:
This section feels like it is getting quite long for a new Julia user Catalyst-focused tutorial. Is all this detail really needed right up front? It seems like it is making the tutorial take a lot longer to getting to actually using Catalyst. Maybe some of it can be moved to the end of the tutorial into a section with more details on package management, and the beginning part can just show how to make an environment and install/use Catalyst?

Reply:
Thanks for the feedback.

I agree. It is a lot stuff to go through before evening seeing a model. Having a really minimal version, and then referencing a full explanation at the end of the tutorial seems like a good solution. I will rewrite using this approach.